### ISSUE TITLE
[RLlib] MAML does not work with TF2 in Ray 2.3.1

### ISSUE THREAD

**Issue by simonsays1980** [ray-project/ray#34620]
### What happened + What you expected to happen

# What happened
I ran the example script shown below with all three frameworks, while with TF1 and Torch MAML runs, it does not so for TF2. 

The problems are: 
1. In TF2 the `_loss_input_dict` in the `initialize_loss_with_dummy_batch()` does not exist and so the `split` key is not there. 
2. Solving this issue in the same way the `MAMLTorchPolicy` does it, gives another error coming from `tf.split` because 
       - the `tf1.GraphKeys.TRAINABLE_VARIABLES` is empty for TF2
       - the `self.policy_vars` do not contain the `log_std_var` from the model and therefore the logits are too "short". 
3. Solving this error by
       - receiving variables via `self.model.base_model.trainable_weights` and 
       - checking for the `log_std_var` in the policy model and adding it to the variable  
    results in another one as `tf.gradients` do not exist for TF2 and the `GradientTape`needs to be used there.
 
# What you expected to happen
That the example below runs with all three frameworks.

### Versions / Dependencies

Ray 2.3.1
Python 3.9.12
Fedora Linux 37

### Reproduction script

```python
from ray.rllib.examples.env.pendulum_mass import PendulumMassEnv
from ray.rllib.algorithms.maml.maml import MAMLConfig

from ray.tune import register_env
from ray import air, tune

register_env("pendulum_mass", lambda config: PendulumMassEnv())

config = (
    MAMLConfig()
    .environment(
        env="pendulum_mass",
        clip_actions=False,
    )
    .rollouts(
        rollout_fragment_length=200,
        num_rollout_workers=2,
        num_envs_per_worker=4,                    
    )
    .framework(
        framework="tf2",
        eager_tracing=False,
    )
    .training(
        inner_adaptation_steps=1,
        maml_optimizer_steps=5,
        gamma=0.99,
        lambda_=1.0,
        lr=0.001,
        vf_loss_coeff=0.5,
        clip_param=0.3,
        kl_target=0.1,
        kl_coeff=0.001,
        inner_lr=0.03,
        model={
            "fcnet_hiddens": [64, 64],
            "free_log_std": True,
        }        
    )
    .exploration(
        explore=True,
    )
)
import ray
ray.init(local_mode=True)
tuner = tune.Tuner(
    "MAML",
    param_space=config.to_dict(),
    run_config=air.RunConfig(
        stop={"training_iteration": 10}
    )
)

tuner.fit()
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by ArturNiederfahrenhorst** [comment]
@simonsays1980 Thanks for the clear description of the problem.
Sorry for not catching this on our side. We simply don't support TF2 here.
We'd be happy to accept a contribution that fixes it. Would you be willing to make such a contribution? Otherwise, I'll put up a PR that raises a more informative error when attempting to run MAML with tf2.

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!

**Comment by simonsays1980** [comment]
@cszhu Yeah makes sense. We deprecate this stack in Q3 and I am working on a new stack MAML.
